feat(ui): Begin work on robot settings (switches/speaker/map data management)

ccoors opened the pull request:
## Speaker and persistent data control

Type A:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] UI Feature
- [ ] Refactor/Code Cleanup
- [ ] Docs
- [ ] Capability implementation for existing core capability
- [ ] New robot implementation
  
Type B:
- [ ] New capability
- [ ] New core feature

# Description (Type A)

Also contains some refactoring to remove code duplications.

![image](https://user-images.githubusercontent.com/8963459/133906924-f92be681-7f13-4b7f-943c-ac8f292c3762.png)

Hypfer commented on frontend/src/robot/capabilities/MapControl.tsx:
```diff
@@ -0,0 +1,111 @@
+                </Typography>
+                <Divider/>
+                <Typography variant="body2">
+                    Persistent data is a feature of some robots which allows to save no-go areas and virtual walls. It
```
This text might be slightly misleading as there are robots (e.g. Dreame vSLAM ones) where you can't disable persistent data. It's always enabled

Hypfer commented on frontend/src/robot/capabilities/MapControl.tsx:
```diff
@@ -0,0 +1,111 @@
+                    also allows the robot to drive back to the dock wherever it is and keeps the map from being rotated.
+                </Typography>
+                <PersistentMapControl supported={persistentMapControl}/>
+                {mapReset && <MapResetControl/>}
```
It might make sense to rethink this and decouple the map reset from the persistent data toggle.

The map reset should probably be relocated to an area with all relevant map management functions while the persistent map toggle should be moved to where all the other toggles (e.g. KeyLock, Carpet Mode, etc) are found

ccoors commented:
You're right. I probably shouldn't write code that late :/

I can't get back to this for a week or so, if you want to get Timers and Events out to people now might be a suitable time for a release... ? ;)

ccoors commented:
Okay, updated stuff and added the other toggles:

![image](https://user-images.githubusercontent.com/8963459/134770816-9fa99176-7320-45ca-8231-c8922ddff081.png)
